src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.tsx

```diff
@@ -16,44 +17,38 @@ export interface TooltipOverlayProps {
   preferredPosition?: PositionedOverlayProps['preferredPosition'];
   children?: React.ReactNode;
   activator: HTMLElement;
+  accessibilityLabel?: string;
   onClose(): void;
 }
 
-export class TooltipOverlay extends PureComponent<TooltipOverlayProps, never> {
-  render() {
-    const markup = this.props.active ? this.renderOverlay() : null;
+export function TooltipOverlay({
+  active,
+  activator,
+  preferredPosition = 'below',
+  preventInteraction,
+  id,
+  children,
+  light,
+  accessibilityLabel,
+}: TooltipOverlayProps) {
+  const i18n = useI18n();
+  const markup = active ? (
+    <PositionedOverlay
+      active={active}
+      activator={activator}
+      preferredPosition={preferredPosition}
+      preventInteraction={preventInteraction}
+      render={renderTooltip}
+    />
+  ) : null;
 
-    return markup;
-  }
-
-  // eslint-disable-next-line @shopify/react-no-multiple-render-methods
-  private renderOverlay = () => {
-    const {
-      active,
-      activator,
-      preferredPosition = 'below',
-      preventInteraction,
-    } = this.props;
+  return markup;
 
-    return (
-      <PositionedOverlay
-        active={active}
-        activator={activator}
-        preferredPosition={preferredPosition}
-        preventInteraction={preventInteraction}
-        render={this.renderTooltip}
-      />
-    );
-  };
-
-  // eslint-disable-next-line @shopify/react-no-multiple-render-methods
-  private renderTooltip: PositionedOverlayProps['render'] = (
-    overlayDetails,
-  ) => {
+  function renderTooltip(
```

Comment:
Should this be wrapped in a useCallback so we don't redefine the function every time?

Reply:
I left it as is because positioned overlay props are unlikely to be pure. Don't have a strong opinion here though, happy to make the change if you think it's valuable! @BPScott

Reply:
I'm going on vacation but if you're ok with this would you mind hitting squash and merge for me so other teams aren't blocked 🙏